### PR TITLE
Add `undefined-loop-variable` example

### PR DESCRIPTION
**Pull request by harupy**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Refs #5953

Closes #XXXX

### CODE REVIEW

File: doc/data/messages/u/undefined-loop-variable/bad.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+def find_even_number(lst):
+    for x in lst:
+        if x % 2 == 0:
+            break
+
+    return x  # [undefined-loop-variable]
```
Comment (harupy) [inline doc/data/messages/u/undefined-loop-variable/bad.py:6]: pylint raises two duplicate errors against line 6. Is this a bug?

```
> pylint doc/data/messages/u/undefined-loop-variable/bad.py
************* Module bad
doc/data/messages/u/undefined-loop-variable/bad.py:6:11: W0631: Using possibly undefined loop variable 'x' (undefined-loop-variable)
doc/data/messages/u/undefined-loop-variable/bad.py:6:11: W0631: Using possibly undefined loop variable 'x' (undefined-loop-variable)

------------------------------------------------------------------
Your code has been rated at 6.00/10 (previous run: 6.00/10, +0.00)
```

Comment (harupy) [reply doc/data/messages/u/undefined-loop-variable/bad.py:6]:
```suggestion
def find_even_number(lst):
    for x in lst:
        if x % 2 == 0:
            break

    return x  # [undefined-loop-variable,undefined-loop-variable]
```

can make the test pass.

Comment (Pierre-Sassoulas) [reply doc/data/messages/u/undefined-loop-variable/bad.py:6]: This look like an issue, what do you think @jacobtylerwalls ?

Comment (jacobtylerwalls) [reply doc/data/messages/u/undefined-loop-variable/bad.py:6]: Let's hold this PR on the hope we can merge a fix relatively soon.

Comment (Pierre-Sassoulas) [reply doc/data/messages/u/undefined-loop-variable/bad.py:6]: @jacobtylerwalls fixed that on the main branch in #6925, could you merge main or rebase please @harupy ?

Comment (harupy) [reply doc/data/messages/u/undefined-loop-variable/bad.py:6]: rebased

File: doc/data/messages/u/undefined-loop-variable/good.py [hunk 2]
```diff
@@ -1 +1,4 @@
-# This is a placeholder for correct code for this message.
+def find_even_number(lst):
+    for x in lst:
+        if x % 2:
+            return x
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/u/undefined-loop-variable/good.py:4]:
```suggestion
def find_even_number(numbers):
    for x in numbers:
        if x % 2:
            return x
    return None            
```

File: doc/data/messages/u/undefined-loop-variable/bad.py (hunk 1 above)
Comment (Pierre-Sassoulas) [inline doc/data/messages/u/undefined-loop-variable/bad.py:6]:
```suggestion
def find_even_number(numbers):
    for x in numbers:
        if x % 2 == 0:
            break
    return x  # [undefined-loop-variable]
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
I think even if inconsistent return is not activated it's better to be explicit, let me know what you think.

**Pull request merged**